astropy/io/votable/tree.py

```diff
@@ -1115,22 +1115,25 @@ def to_table_column(self, column):
         column.meta['values'] = meta
 
     def from_table_column(self, column):
-        if 'values' not in column.meta:
-            return
-
-        meta = column.meta['values']
-        for key in ['ID', 'null']:
-            val = meta.get(key, None)
-            if val is not None:
-                setattr(self, key, val)
-        if 'min' in meta:
-            self.min = meta['min']['value']
-            self.min_inclusive = meta['min']['inclusive']
-        if 'max' in meta:
-            self.max = meta['max']['value']
-            self.max_inclusive = meta['max']['inclusive']
-        if 'options' in meta:
-            self._options = list(meta['options'].items())
+        if isinstance(column, Quantity):
+            column = column.info
+        if column.meta is not None:
```

Comment:
Write as:
```
if column.info.meta is None or 'values' not in column.info.meta:
    return
```
That way you don't make a big diff below by putting the original code into a new block.  This is something to strive for (minimizing diffs) because it makes the code review easier.